### PR TITLE
Update user boot scripts deployment to include injection of $ATMO_ variables

### PR DESCRIPTION
**Pull request by steve-gregory**
This first PR include support for a single variable, $ATMO_USER that represents the user of the identity that created the instance.
Any other variables injected should also carry the $ATMO_ namespace.

Scripts that execute after this should be expected to "source the file $HOME/.bashrc for the deploying user (Usually root)".

A later version would ideally have ansible support. Will consult with @amercer1 about development on this front.

This will be patched into production to help a user get their image up and running properly.

### CODE REVIEW

File: core/models/boot_script.py [hunk 1]
```diff
@@ -61,8 +61,8 @@ def get_title_slug(self):
         return slugify(self.title).replace('-', '_')
 
     def get_text(self):
-        if self.script_type.name == 'Full Text':
-            return self.script_text
+        if self.script_type.name == 'Raw Text':
+            return self.script_text.strip ()  # Remove whitespace
```
Comment (lenards) [inline core/models/boot_script.py]: `whitespace == ` :japanese_goblin:

File: service/deploy.py [hunk 2]
```diff
@@ -464,9 +465,27 @@ def wrap_script(script_text, script_name):
...
+        """#!/bin/bash -x
+atmo_user="%s"
+env_file="%s"
+env_vars=`cat $env_file`
+if [[ $env_vars == *"ATMO_USER="* ]]; then
+    sed -i "s/ATMO_USER=.*/ATMO_USER=\"$atmo_user\"/" $env_file
+else
+    echo "ATMO_USER=\"$atmo_user\"" >> $env_file
+fi
+""" % (username, "$HOME/.bashrc")
```
Comment (lenards) [inline service/deploy.py]: I agree with the `TODO` here - we might want to make this a template, similarly to how we handle rendering into the [email feedback](https://github.com/iPlantCollaborativeOpenSource/atmosphere/tree/master/core/templates/core/email), etc. 

Maybe `core/templates/scripts`...

File: service/tasks/driver.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
 from core.models.identity import Identity
 from core.models.profile import UserProfile
 
-from service.deploy import check_process, wrap_script, echo_test_script,\
+from service.deploy import _inject_env_script, check_process, wrap_script, echo_test_script,\
```
Comment (lenards) [inline service/tasks/driver.py]: Maybe I'm missing the mark on the purpose of `_method_name` - but when I see an underbar method imported - that's a signal for me to pay extra attention. I believe that with `_` preview we're saying that `_inject_env_script` is _intended_ to be use *within* `service.deploy` - but not outside. 

I know we need to forge ahead ... or [_ford the river_](http://coldfrontmag.com/wp-content/uploads/2013/06/2-Trail_069.jpg). But just something I _note_ as I look at code. 

Let me know if I'm off-base.

File: service/tasks/driver.py [hunk 4]
```diff
@@ -557,9 +563,9 @@ def get_remove_status_chain(driverCls, provider, identity, instance):
     return start_chain
 
 
-def get_chain_from_build(driverCls, provider, identity, instance,
-                         username=None, password=None, redeploy=False,
-                         deploy=True):
+def get_chain_from_build(
+        driverCls, provider, identity, instance, core_identity,
+        username=None, password=None, redeploy=False, deploy=True):
```
Comment (lenards) [inline service/tasks/driver.py]: :+1: on the re-format.

File: service/tasks/driver.py [hunk 5]
```diff
@@ -589,9 +595,9 @@ def print_chain(start_task, idx=0):
         mystr += print_chain(task, idx+1)
     return mystr
 
-def get_chain_from_active_no_ip(driverCls, provider, identity, instance,
-                                username=None, password=None, redeploy=False,
-                                deploy=True):
+def get_chain_from_active_no_ip(
+        driverCls, provider, identity, instance, core_identity,
+        username=None, password=None, redeploy=False, deploy=True):
```
Comment (lenards) [inline service/tasks/driver.py]: :+1:

File: service/tasks/driver.py [hunk 6]
```diff
@@ -613,16 +619,16 @@ def get_chain_from_active_no_ip(driverCls, provider, identity, instance,
...
     end_chain.link(deploy_start)
     return start_chain
 
 
-def get_chain_from_active_with_ip(driverCls, provider, identity, instance,
-                                  username=None, password=None,
-                                  redeploy=False, deploy=True):
+def get_chain_from_active_with_ip(
+        driverCls, provider, identity, instance, core_identity,
+        username=None, password=None, redeploy=False, deploy=True):
```
Comment (lenards) [inline service/tasks/driver.py]: :+

File: service/tasks/driver.py [hunk 7]
```diff
@@ -847,6 +867,9 @@ def destroy_instance(instance_alias, user, core_identity_uuid):
       max_retries=10)
 def deploy_script(driverCls, provider, identity, instance_id,
                   script, **celery_task_args):
+    """
+    #TODO: Should this be ansible now?
+    """
```
Comment (lenards) [inline service/tasks/driver.py]: This is a question I _had_ too. I'm not sure about the answer. But it seems like a reasonable division of labor.

**Comment by lenards** [comment]
Overall - I think you are _good to go_ here. 

I think we want to consider: 
- [ ] finding a home for `_inject_env_script` in a more fitting place (maybe template?)
- [ ] determine how this _could_ be handled by `atmosphere-ansible`.

**Comment by lenards** [comment]
But, :shipit:

**Comment by lenards** [comment]
I see this is still "Open" - but I'm okay with getting this in now. :ship:

File: service/tasks/driver.py (hunk 3 above)
Comment (lenards) [inline service/tasks/driver.py]: If we _intend_ for `_inject_env_script` to be used outside of the defining file, I would change this to `inject_env_script`.

File: core/models/boot_script.py (hunk 1 above)
Comment (steve-gregory) [inline core/models/boot_script.py]: The irony here...

File: service/tasks/driver.py (hunk 3 above)
Comment (steve-gregory) [inline service/tasks/driver.py]: Noted for fixing

**Comment by steve-gregory** [comment]
Updated the essentials. Will leave TODO's in the PR so that they can be addressed by @amercer1 at a later date.

**Comment by lenards** [comment]
I think this is ready. You got the :shipit: from me.

**Comment by lenards** [comment]
Bon voyage :ship:

**Comment by steve-gregory** [comment]
Will merge into prod tonight.

**Comment by lenards** [comment]
Okay.

**Pull request merged**